### DISCUSSION TITLE
Feature request: support Parquet INT96 data type as the target for DuckDB TIMESTAMP data type

### DISCUSSION THREAD

**Discussion by orthoxerox** [duckdb/duckdb#5512] [General]
The work done in #707 allows DuckDB to read and correctly interpret Hive/Impala timestamps. However, it is impossible to generate a parquet file that can be read by Hive from DuckDB: DuckDB exports timestamps as INT64 with `TimestampType(isAdjustedToUTC=0, unit=TimeUnit(MILLIS=<null>, MICROS=MicroSeconds(), NANOS=<null>))` as the column's logical type, while Hive expects it to be INT96 and fails with `Error in fetching data rows: *org.apache.hive.service.cli.HiveSQLException:java.io.IOException: org.apache.hadoop.hive.ql.metadata.HiveException: java.lang.ClassCastException: org.apache.hadoop.io.LongWritable cannot be cast to org.apache.hadoop.hive.serde2.io.TimestampWritableV2:14:13;`

Looks like function `TimestampToImpalaTimestamp` already exists, but `parquet_writer.cpp` always uses INT64.

**Comment by hannes** [comment]
Would that not be a bug in Hive?

**Reply by orthoxerox** [reply]
Not a bug per se. While INT96 is deprecated in Parquet (https://github.com/apache/parquet-format/blob/master/Encodings.md), timestamps are still mapped to it in Hive by default, and it's documented in [the manual](https://cwiki.apache.org/confluence/display/Hive/LanguageManual+Types#LanguageManualTypes-TimestampstimestampTimestamps). INT64 timestamps with logical type mappings will be supported only in Hive 4.0
